modules/masterbar/masterbar.php

```diff
@@ -68,6 +118,50 @@ function __construct() {
 		add_action( 'wp_logout', array( $this, 'maybe_logout_user_from_wpcom' ) );
 	}
 
+	public function tracks_and_redirect_add_route() {
+		add_rewrite_rule( 'jetpack-track-and-bounce.php\?tracks_and_bounce=(\w*)', 'index.php?tracks_and_bounce=$matches[1]', 'top' );
+	}
+
+	public function tracks_and_redirect_add_query_vars( $query_vars ) {
+		$query_vars[] = 'tracks_and_bounce';
+		$query_vars[] = 'tracks_and_bounce_nonce';
+		return $query_vars;
+	}
+
+	public function tracks_and_redirect_parse_request( &$wp ) {
+		if (
+			array_key_exists( 'tracks_and_bounce', $wp->query_vars ) &&
+			array_key_exists( $_GET[ 'tracks_and_bounce' ], $this->links_tracks_events ) &&
+			wp_verify_nonce( $_GET[ 'tracks_and_bounce_nonce' ], 'jp-tracks-masterbar-nonce' )
+		) {
+			$target = $this->tracks_and_redirect_get_target( $_GET[ 'tracks_and_bounce' ] );
+
+			JetpackTracking::record_user_event(
+				$this->links_tracks_events[ $_GET[ 'tracks_and_bounce' ] ][0],
+				$data = array( 'target' => $target )
+			);
+			wp_redirect( $target );
```

Comment:
I know that we are atm building this link and it looks safe atm, but could we switch this to `wp_safe_redirect` to avoid accidentally exposing open redirect vulnerability in the future?

Reply:
Excellent point. Updated on 3f119d6f2195812eb97a8c8343d1b5a7c18e61b1